From d8b71be0fa3f7c5feea9135c7552fd83e8e48d9b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Wed, 20 Mar 2019 08:30:25 +0200
Subject: [PATCH 481/741] crypto: caam - fix building job ring as module
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit 9335f72011fe75f77b1e47c6c4e243ead7f6c05b from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

In case job ring is built as module, the shared descriptors libraries
also have to be built as modules - otherwise the linking will fail:
[...]
ERROR: "cnstr_shdsc_rfc4106_encap" [drivers/crypto/caam/caam_jr.ko] undefined!
ERROR: "cnstr_shdsc_aead_null_decap" [drivers/crypto/caam/caam_jr.ko] undefined!
ERROR: "cnstr_shdsc_gcm_encap" [drivers/crypto/caam/caam_jr.ko] undefined!
[...]

Fixes: a9ade076ed8b ("crypto: caam - duplicate files linking (kernel 4.14 sync)")
Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/crypto/caam/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/Kconfig b/drivers/crypto/caam/Kconfig
index 0e014b834cc0..3de275c95fb7 100644
--- a/drivers/crypto/caam/Kconfig
+++ b/drivers/crypto/caam/Kconfig
@@ -3,10 +3,10 @@ config CRYPTO_DEV_FSL_CAAM_COMMON
 	tristate
 
 config CRYPTO_DEV_FSL_CAAM_CRYPTO_API_DESC
-	bool
+	tristate
 
 config CRYPTO_DEV_FSL_CAAM_AHASH_API_DESC
-	bool
+	tristate
 
 config CRYPTO_DEV_FSL_CAAM
 	tristate "Freescale CAAM-Multicore platform driver backend"
-- 
2.17.1

